### PR TITLE
Enable NFT Pinning by default

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/29017

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/common/features.cc [hunk 1]
```diff
@@ -50,7 +50,7 @@ BASE_FEATURE(kBraveWalletSnsFeature,
 
 BASE_FEATURE(kBraveWalletNftPinningFeature,
              "BraveWalletNftPinning",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
```
Comment (yrliou) [inline components/brave_wallet/common/features.cc:53]: Should it stay disabled for Android?

Comment (cypt4) [reply components/brave_wallet/common/features.cc]: Fixed

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.52.17 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | `bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}`
OS | Windows 11 Version 22H2 (Build 22621.1485)

Verified test plan from [brave-browser#29017](https://github.com/brave/brave-browser/issues/29017)

<details>
<summary>Enabling the feature - <b>Passed</b></summary>
<br>

- Verified `brave://flags/#enable-nft-pinning` is set to `Default` 
- Verified adding an NFT to visible asset list, automatically shows the NFT pinning banner 
- Verified clicking on `Learn more` on the banner loads interstitial page `brave://wallet/crypto/local-ipfs-node`
- Verified clicking on `Check which NFT's of mine can be pinned` loads second interstitial page `brave://wallet/crypto/inspect-nfts`
- Verified clicking on `Keep my NFTs always online` starts IPFS on localhost and start pinning the available NFT's
- Verified when `Keep my NFTs always online` is selected, setting `Automatically pin NFT's` is enabled, default is disabled

</details>
<details>
<summary>Pinning - <b>Passed</b></summary>
<br>

- Verified able to pin NFTs
- Verified when pinning is in process, shows message banner message about being pinned in local node
- Verified when successfully pinned, show message about number of NFTs that are pinned
- Additional verification notes for [banner messages](https://github.com/brave/brave-core/pull/17557#issuecomment-1496988390) and [pinning data recursively](https://github.com/brave/brave-core/pull/17742#issuecomment-1497143159) 

</details>
<details>
<summary>Unpinning - <b>Passed</b></summary>
<br>

- Verified after removing an NFT and running `curl http://127.0.0.1:<api port could be viewed at chrome://ipfs-internal>/api/v0/pin/ls?type=recursive`, shows `{"Keys":{}}` message 
image

</details>
<details>
<summary>Pins Validation - <b>Passed</b></summary>
<br>

- Verified deleting `brave_ifps` folder and moving date ahead and restarting the browser restores `brave_ipfs` folder again
- Verified resetting device date back to current date after restart starts to pin the NFT's again.
- Running `curl http://127.0.0.1:<api port could be viewed at chrome://ipfs-internal>/api/v0/pin/ls?type=recursive` continues to show the pinned NFT details
</details>
<details>
<summary>Disable feature- <b>Passed</b></summary>
<br>

- Verified disabling `Automatically pin NFTs` setting and reloading the NFT tab shows the onboarding banner message
- Verified manually deleting the NFT's from visible asset list also removes the onboarding banner message
</details>
<details>
<summary>Cleanup- <b>Passed</b></summary>
<br>

- Verified disabling `Automatically pin NFTs` setting, a new setting for `Clear pinned NFTs` is shown
- Verified clicking that clears the NFTs pinned state and shows onboarding banner message
- Encountered [#29553](https://github.com/brave/brave-browser/issues/29553) & [#29554](https://github.com/brave/brave-browser/issues/29554)
</details>
<details>
<summary>Local node behaviour- <b>Passed</b></summary>
<br>

- Verified when user click on `Keep my NFTs always online` from interstitial page, IPFS local node is installed and started.
</details>
<details>
<summary>Error handling- <b>Passed</b></summary>
<br>

- Verified if NFT's `content-type` is not of the format `image/*`, NFT pinning fails. Verified as part of [brave-browser#28775 (comment)](https://github.com/brave/brave-browser/issues/28775#issuecomment-1475924572)
- Verified due to network issues NFT pinning fails, it starts the pinning process again when browser is restarted
</details>

**Pull request merged**